### DISCUSSION TITLE
Language service: triggering automatic definition creation in implementation file

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#8597] [Q&A] [answered]
Hello,

Consider `Main.cpp`:

    //Main.cpp
    #include "Main.h"
    
    int main(){
    	class SIMPLE smpl;
    	smpl.setval(10);
    	smpl.setname();
    	smpl.printname();
    }

and `Main.h` thus:

    //Main.h
    #pragma once
    #include <iostream>
    #include <string>
    #include <vector>
    class SIMPLE {
    public:
    	int retval() const { return val; }
    	void setval(int i) { val = i; }
    	void setname();
    	void printname();
    private:
    	int val;
    	std::string name;
    };

The member functions `setname()` and `printname()` are missing definitions. VSCode Intellisense does not seem to be able to warn the user about this. See figure below which does not have any squiggles:

![vscode](https://user-images.githubusercontent.com/70434658/147105395-67b8916f-90d7-40d9-acb5-db171d6dcb98.jpg)

Visual Studio IDE, on the other hand, is able to point out the missing definitions via squiggles and allows for creation of the definition in the implementation file directly from the header file. See gif image below:

![helperfunctions](https://user-images.githubusercontent.com/70434658/147105476-15a9e576-dd32-41d5-81ca-98683058bd44.gif)

I can understand that VSIDE takes more memory and perhaps that is why it is able to provide this ability. I was curious to know if this is at all available in VSCode currently via any setting, perhaps, that I can tweak.

Thank you.

**Comment by sean-mcmanus** [comment, marked as answer]
This is already being worked on for our 1.9.0 release (or so). You can track the progress at https://github.com/microsoft/vscode-cpptools/issues/664 .
